server/store/peer_store.go

```diff
@@ -24,6 +24,12 @@ type PeerStore interface {
 	GraduateLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
 	AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWant int) (peers, peers6 []chihaya.Peer, err error)
 	CollectGarbage(cutoff time.Time) error
+
+	GetSeeders(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
+	GetLeechers(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
```

Comment:
This seems logical for now, but for example Facebook's subnet middleware is going to want to know the client's IP address to pick out the seeders. Maybe we should change all of these methods to use `AnnounceRequest` or `ScrapeRequest`.

Thoughts? @pushrax @mrd0ll4r